pkg/cloud/scope/machine.go

```diff
@@ -370,6 +372,41 @@ func (m *MachineScope) IsEKSManaged() bool {
 	return m.InfraCluster.InfraCluster().GetObjectKind().GroupVersionKind().Kind == ekscontrolplanev1.AWSManagedControlPlaneKind
 }
 
+func (m *MachineScope) IsControlPlaneExternallyManaged() bool {
```

Comment:
Honestly, I'm not so much happy about this solution, since several steps are returning errors, and we should avoid that at the beginning for the scope.

I'd suggest introducing a new function to the interface `ClusterScoper`,  such as `ControlPlane`, which returns the Control Plane object in order to check its status if it is externally managed.

Reply:
Addressed with changes from be4b28aff.